tls/s2n_client_extensions.c

```diff
@@ -429,3 +445,17 @@ static int s2n_recv_client_sct_list(struct s2n_connection *conn, struct s2n_stuf
     /* Skip reading the extension, per RFC6962 (3.1.1) it SHOULD be empty anyway  */
     return 0;
 }
+
+
+static int s2n_recv_client_max_frag_len(struct s2n_connection *conn, struct s2n_stuffer *extension)
```

Comment:
I don't think we should honor the extension by default in server mode. I think we should add an API to opt-in. Existing server deployments of s2n would not want this enabled suddenly.